omc3/harpy/clean.py

```diff
@@ -169,44 +172,73 @@ def _resync_bpms(harpy_input, bpm_data, model):
     return bpm_data.iloc[:, :-1]
 
 
-def svd_decomposition(bpm_data, num_singular_values):
+def svd_decomposition(bpm_data, num_singular_values, dominance_limit=None):
     """
     Computes reduced (n largest values) singular value docomposition of a matrix (bpm_data)
 
     Args:
         bpm_data: matrix to be decomposed
         num_singular_values: input options object that contains
+        dominance_limit: limit on SVD dominance
 
     Returns:
-        An indexed DataFrame of U matrix, product of S and V^T martices, and mean of original matrix
+        An indexed DataFrame of U matrix, product of S and V^T martices,
+            mean of original matrix and U matrix mask for cleaned elements
     """
     bpm_data_mean = bpm_data.to_numpy().mean()
-    u_mat, svt_mat = _get_decomposition(bpm_data - bpm_data_mean, num_singular_values)
-    return pd.DataFrame(index=bpm_data.index, data=u_mat), svt_mat, bpm_data_mean
+    u_mat, svt_mat, u_mat_mask = _get_decomposition(bpm_data - bpm_data_mean, num_singular_values,
+                                                    dominance_limit=dominance_limit)
+    return pd.DataFrame(index=bpm_data.index, data=u_mat), svt_mat, bpm_data_mean, u_mat_mask
 
 
-def _get_decomposition(matrix, num):
+def _get_decomposition(matrix, num, dominance_limit=None):
     """
     Removes noise floor
     Requiring K singular values from MxN matrix results in matrices sized: ((MxK) x diag(K) x (K,N))
 
     Returns:
-        U (MxK),  SVt (diag(K).(K,N))
+        U (MxK),  SVt (diag(K).(K,N)), U matrix mask for cleaned elements (same dimensions as U)
     """
     u_mat, s_mat, vt_mat = np.linalg.svd(matrix / np.sqrt(matrix.shape[1]), full_matrices=False)
+
+    if dominance_limit is not None:
```

Comment:
Check if in function, to have u_mat _mask created only there (i.e. return unchanged s_mat and u_mat)